src/Tizen.Sensor/Tizen.Sensor/Sensor.cs

```diff
@@ -541,9 +689,778 @@ private bool CheckSensorHandle()
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Provider handle does not exist");
+            }
         }
+
+        /// <summary>
+        /// Registers the callback function to be invoked when a sensor listener stops the sensor provider.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        private void SetProviderStopCb()
+        {
+            if (CheckProviderHandle())
+            {
+                _sensorProviderStopCallBack = (IntPtr _provider, Int64 _userData) =>
+                {
+                };
+
+                int error = Interop.SensorProvider.SetProviderStopCb(_sensorProviderHandle, _sensorProviderStopCallBack, IntPtr.Zero);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error in setting stop callback of sensor provider");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderSetStopCb Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Provider handle does not exist");
+            }
+        }
+
+        /// <summary>
+        /// Registers the callback function to be invoked when the interval is changed.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        private void SetProviderIntervalChangedCb()
+        {
+            if (CheckProviderHandle())
+            { 
+                _sensorProviderIntervalChangedCallBack = (IntPtr _provider, uint IntervalMs, Int64 _userData) =>
+                {
+                    ProviderIntervalChangedEventArgs e = new ProviderIntervalChangedEventArgs(_provider, IntervalMs, _userData)
+                    {
+                    };
+                    ProviderIntervalChanged?.Invoke(null, e);
+                };
+
+                int error = Interop.SensorProvider.SetProviderIntervalChangedCb(_sensorProviderHandle, _sensorProviderIntervalChangedCallBack, IntPtr.Zero);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error in setting interval callback of sensor provider");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "Sensor.ProviderSetIntervalChangedCb Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Provider handle does not exist");
+            }
+        }
+
+        /// <summary>
+        /// Publishes a sensor event through the declared sensor.
+        /// This function publishes a sensor's data to its listeners.
+        /// </summary>
+        /// <param name="sensorEventS">
+        /// The sensor event.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void ProviderPublish(SensorEventStruct sensorEventS)
```

Comment:
How to create SensorEventStruct? all property setter is private

Reply:
What means of last `S`?

Reply:
typedef struct
{   int accuracy;                 
	unsigned long long timestamp;  
	int value_count;               
	float values[MAX_VALUE_SIZE];  
} sensor_event_s;

In C code they used sensor_event_s for the struct ... I used the same name here

Reply:
It is not c, please follow .net framework guide

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/

Reply:
There are two structures with same name. One is in Interop and it's private. Another one is public and is accessible to user. As private one is used in many places, didn't disturb it. Created Public structure for this specific API. Kindly suggest on this.

Reply:
I recommend, define a new class to represent `SenserEvent` and convert to `Interop.SensorEventStruct` in your implementation